### PR TITLE
rtnetlink: implement MPLSNextHop encoding and decoding

### PR DESCRIPTION
**Pull request by mdlayher**
Seems to work and the fuzzer doesn't crash.

Note the new test: it became such a hassle to keep updating the various length and type fields that I don't think it's worth extending the unmarshaling test any further. After experimenting on a system with MPLS routes, I can confirm this logic seems to work.

### CODE REVIEW

File: route.go [hunk 1]
```diff
@@ -437,6 +456,103 @@ func (nh *NextHop) decode(ad *netlink.AttributeDecoder) error {
...
+
+// encodeEncap encodes netlink attribute values related to encapsulation from
+// a NextHop.
+func (nh *NextHop) encodeEncap(ae *netlink.AttributeEncoder) error {
+	// TODO: this only handles MPLS encapsulation as that is all we support.
+
+	// Allocate enough space for an MPLS label stack.
+	var (
+		i int
+		b = make([]byte, 4*len(nh.MPLS))
```
Comment (bradbeam) [inline route.go:489]: I feel like that `4` is used in a few places, it might be worth making it a const for mplsLabelSize

Comment (jsimonetti) [reply route.go:489]: 4 bytes is actually the entire MPLS header. Calling it mplsLabelSize would be very inconsistent as the label is only a portion of those 4 bytes. If this is at all needed (which can be argued for clearity sake) I would call it mplsHeaderSize

Comment (jsimonetti) [reply route.go:489]: 4 bytes is actually the entire MPLS header. Calling it mplsLabelSize would be very inconsistent as the label is only a portion of those 4 bytes. If this is at all needed (which can be argued for clearity sake) I would call it mplsHeaderSize

**Review by jsimonetti — APPROVED** [review]
LGTM

**Comment by jsimonetti** [comment]
> Note the new test: it became such a hassle to keep updating the various length and type fields that I don't think it's worth extending the unmarshaling test any further. After experimenting on a system with MPLS routes, I can confirm this logic seems to work.

I can appreciate the problem. I don't see any harm in this. If we ever get a bug we can always add a regression test for it.

**Pull request merged**